src/components/ui/dashboard/join-coaching-session.tsx

```diff
@@ -0,0 +1,153 @@
+import React from "react";
+import { useAppStateStore } from "@/lib/providers/app-state-store-provider";
+import { Id } from "@/types/general";
+import { DynamicApiSelect } from "./dynamic-api-select";
+import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
+import { Organization } from "@/types/organization";
+import { CoachingRelationshipWithUserNames } from "@/types/coaching_relationship_with_user_names";
+import {
+  CoachingSession,
+  getCoachingSessionById,
+} from "@/types/coaching-session";
+import { DateTime } from "ts-luxon";
+import { Label } from "@/components/ui/label";
+import { Button } from "../button";
+import Link from "next/link";
+import { fetchOrganization } from "@/lib/api/organizations";
+import { fetchCoachingRelationshipWithUserNames } from "@/lib/api/coaching-relationships";
+import { fetchCoachingSessions } from "@/lib/api/coaching-sessions";
+
+export interface CoachingSessionCardProps {
+  userId: Id;
+}
+
+export function JoinCoachingSession({
+  userId: userId,
+}: CoachingSessionCardProps) {
+  const { setOrganizationId } = useAppStateStore((state) => ({
+    setOrganizationId: state.setOrganizationId,
+  }));
+  const { setRelationshipId } = useAppStateStore((state) => ({
+    setRelationshipId: state.setRelationshipId,
+  }));
+  const { setCoachingSessionId } = useAppStateStore((state) => ({
+    setCoachingSessionId: state.setCoachingSessionId,
+  }));
+  const { organization, setOrganization } = useAppStateStore((state) => ({
+    organization: state.organization,
+    setOrganization: state.setOrganization,
+  }));
+  const { relationship, setRelationship } = useAppStateStore((state) => ({
+    relationship: state.coachingRelationship,
+    setRelationship: state.setCoachingRelationship,
+  }));
+  const { coachingSession, setCoachingSession } = useAppStateStore((state) => ({
+    coachingSession: state.coachingSession,
+    setCoachingSession: state.setCoachingSession,
+  }));
+  const FROM_DATE = DateTime.now().minus({ month: 1 }).toISODate();
+  const TO_DATE = DateTime.now().plus({ month: 1 }).toISODate();
+
+  const handleOrganizationSelection = (id: Id) => {
+    fetchOrganization(id)
```

Comment:
@qafui This is what I'm referring to, can we/should we rework fetchOrganization to use your use-api-data in this PR, or do that across the board in a second fast-follower PR? I'm open to either option.